SMPyBandits/Policies/SWA.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+""" The Sliding Window Average policy for rotting bandits.
+Reference: [Levine , Crammer & Mannor, 2017].
+"""
+from __future__ import division, print_function  # Python 2 compatibility
+
+__author__ = "Julien Seznec"
+__version__ = "0.1"
+
+from math import sqrt, log
+import numpy as np
+import pandas as pd
+np.seterr(divide='ignore')  # XXX dangerous in general, controlled here!
+
+try:
+    from .IndexPolicy import IndexPolicy
+except ImportError:
+    from IndexPolicy import IndexPolicy
+
+
+class SWA(IndexPolicy):
+    """ The Sliding Window Average policy for rotting bandits.
+    Reference: [Levine , Crammer & Mannor, 2017].
+    """
+    def __init__(self, nbArms, horizon=1, subgaussian=1, maxDecrement=1,alpha=0.2, doublingTrick=False):
+        super(SWA, self).__init__(nbArms)
+        self.t = 0
+        self.nbArms = nbArms
+        self.arms_history = {}
+        for arm in range(nbArms):
+            self.arms_history[arm] = np.array([])
+        self.armSet = set(range(nbArms))
+        self.horizon = horizon
+        self.alpha = alpha if alpha else (2*maxDecrement)**(-2/3)
```

Comment:
Do you want to say, `if alpha is not None` ? This will fail to accept `alpha=0`, maybe that's not what you wanted

Reply:
yes (though \alpha =0 would be a strange choice)